scripts/add_identifiers_to_existing_preprints.py

```diff
@@ -31,6 +34,30 @@ def add_identifiers_to_preprints():
     logger.info('Finished Adding identifiers to {} preprints.'.format(preprints_without_identifiers.count()))
 
 
+def main(dry=True):
+    # Start a transaction that will be rolled back if any exceptions are un
+    with transaction.atomic():
```

Comment:
Instead of using a transaction, how about preventing the request and database change?

```python
if not dry:
     .. request_identifiers_from_ezid...
     ...set_identifier_values(...)